DEVELOPERS_GUIDE.md

```diff
@@ -0,0 +1,87 @@
+# Microsoft GCToolKit Developers Guide
```

Comment:
Did you intend for this to be part of the commit?

Reply:
/me sighs... I intended to have the doc pushed in a doc branch but it got mixed up in this workflow effort... I'll sort this out by rolling the doc into that branch and resubmitting.